tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -541,34 +557,34 @@ void LaunchScalarReduction(OpKernelContext* ctx, OUT_T out, IN_T in,
 template <typename T, typename Op, typename OUT_T, typename IN_T>
 void LaunchRowReduction(OpKernelContext* ctx, OUT_T out, IN_T in, int num_rows,
                         int num_cols, Op op, T init,
-                        const cudaStream_t& cu_stream) {
+                        const gpuStream_t& cu_stream) {
   if (num_cols < 1024) {
     const int threads_per_block = 128;
     const int warps_per_block = threads_per_block / 32;
     int num_blocks = (num_rows + warps_per_block - 1) / warps_per_block;
 
-    RowReduceKernel<<<num_blocks, threads_per_block, 0, cu_stream>>>(
+    GPU_LAUNCH_KERNEL(RowReduceKernel, dim3(num_blocks), dim3(threads_per_block), 0, cu_stream,
         in, out, num_rows, num_cols, op, init);
     return;
   }
 
   // setup segment offsets with counting and transform iterator
   RowOffset row_offset_op(num_cols);
-  cub::CountingInputIterator<int> counting_iter(0);
-  cub::TransformInputIterator<int, RowOffset, cub::CountingInputIterator<int>>
+  gpuprim::CountingInputIterator<int> counting_iter(0);
+  gpuprim::TransformInputIterator<int, RowOffset, gpuprim::CountingInputIterator<int>>
       transform_iter(counting_iter, row_offset_op);
 
   std::size_t temp_storage_bytes = 0;
   Tensor temp_storage;
   for (int i = 0; i < 2; ++i) {
-    auto success = cub::DeviceSegmentedReduce::Reduce(
+    auto success = gpuprim::DeviceSegmentedReduce::Reduce(
         i == 0 ? nullptr : temp_storage.flat<int8_t>().data(),
         temp_storage_bytes, in, out, num_rows, transform_iter,
         transform_iter + 1, op, init, cu_stream);
 
     OP_REQUIRES(ctx, success == 0,
                 errors::Internal("CUB segmented reduce error",
```

Comment:
the message could use some change i think as it's not solely for CUB now

Reply:
I was wondering about this but left them intact as technically we are using the hip*cub* API as well.